### ISSUE TITLE
Best practices for distributed Keras as tf.estimator

### ISSUE THREAD

**Issue by squared9** [keras-team/keras#10336]
Hello Keras team,

I am searching for a guide outlining best practices for adjusting a Keras model with all accompanying functionality to be able to run it in a meta-parameter (hyperparameter, loss function composition etc.) optimization/orchestration in a distributed fashion and am looking at the recommended way via tf.estimator (which is also employed by AWS' SageMaker that I am evaluating right now).

Converting a model to an estimator seems to be easy; the problem is there is not much written about converting the rest of Keras pipeline to TF. I am working on an ensemble of state-of-art models for certain computer vision problems that seemed to perform exceptionally well on tested domains (Xception, DenseNet-121, DenseNet-201, Wide ResNet from keras_contrib and custom DenseNet-BC). There is a non-trivial generator based on Sequence, the training utilizes Multi-GPU callback, ROC AUC/Matthews correlation coefficients/Cohen's Kappa per epoch callback, learning rate plateau adjustment callback, early stopping callback etc. There are weights provided for each classification class, multiple custom loss functions are defined and the result is also evaluated via CAM heatmaps of activation areas. It might also require some exotic models/functions coming from keras_contrib or other libraries, such as Wide ResNet or PELU. This all obviously works fine as a single Keras training file/notebook.

Is there some general guideline on how to convert these parts to TensorFlow from pure Keras running on TensorFlow backend so that I can run them in a distributed fashion?

Thank you & Best!
